typescript/introduction/tsc-basic-types/tsc-never.md

```diff
@@ -0,0 +1,70 @@
+---
+author: kapnobatai136
+type: normal
+category: must-know
+aspects:
+  - deep
+  - workout
+links:
+  - '[Never Type](https://www.typescriptlang.org/docs/handbook/basic-types.html#never){documentation}'
+
+---
+
+# Never
+
+---
+## Content
+
+The type `never` is used to represent a value that **never** occurs.
+
+Let's take a look at a few examples:
+
+```ts
+function neverEnd(): never {
+  while (true) {}
+}
+
+function throwError(
+  message: string
+): never {
+  throw new Error(message);
+}
+
+function impossibleValue(value) {
```

Comment:
shouldn't this have `function impossibleValue(value): never`?

Reply:
nope.

This function returns `undefined`.

What's impossible is for `value` to be both a `string` and a `number`. That's why it's a `never`.

Reply:
ooo, okay. makes sense now. thanks for clarifying.

I'll resolve it